### PR TITLE
run all integration test on circleci

### PR DESCRIPTION
**Pull request by sunchunqiang**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -173,30 +186,111 @@ jobs:
...
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Polly          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-S3   
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Pinpoint
+      - restore_cache:
+          key: test_result-{{ .Revision }}-IoT          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-KMS_ELB_EC2_Lambda_Lex
```
Comment (kvasukib) [inline .circleci/config.yml]: can you expand the names? Abbreviation makes it hard to understand what tests fail?

File: .circleci/config.yml [hunk 2]
```diff
@@ -173,30 +186,111 @@ jobs:
...
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Pinpoint
+      - restore_cache:
+          key: test_result-{{ .Revision }}-IoT          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-KMS_ELB_EC2_Lambda_Lex                      
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Cognito_Comprehend_Core_Kinesis   
+      - restore_cache:
+          key: test_result-{{ .Revision }}-RG_SES_CW_AW_AG_TL_SNS
```
Comment (kvasukib) [inline .circleci/config.yml]: can you expand the names? Abbreviation makes it hard to understand what tests fail?

File: .circleci/config.yml [hunk 3]
```diff
@@ -173,30 +186,111 @@ jobs:
...
+      - restore_cache:
+          key: test_result-{{ .Revision }}-Transcribe          
+      - restore_cache:
+          key: test_result-{{ .Revision }}-SQS_MAL_MA
```
Comment (kvasukib) [inline .circleci/config.yml]: can you expand the names? Abbreviation makes it hard to understand what tests fail?

File: .circleci/config.yml [hunk 4]
```diff
@@ -135,24 +138,33 @@ commands:
           command: |
             aws configure --profile sdk_s3_release set region us-east-1
             echo -e "[sdk_s3_release]\naws_access_key_id=${S3_RELEASE_AWS_ACCESS_KEY_ID}\naws_secret_access_key=${S3_RELEASE_AWS_SECRET_ACCESS_KEY}\n" >> ~/.aws/credentials
+  pre_start_simulator:
+    description: >-
+      pre start simulator, build may fail is simulator is not started
+    steps:
+      - run:   
+          name: pre-start simulator
+          command: xcrun instruments -w "iPhone 8 (10.1) [" || true
```
Comment (kvasukib) [inline .circleci/config.yml]: is it possible to run on the latest simulator?

File: .circleci/config.yml [hunk 5]
```diff
@@ -135,24 +138,33 @@ commands:
           command: |
             aws configure --profile sdk_s3_release set region us-east-1
             echo -e "[sdk_s3_release]\naws_access_key_id=${S3_RELEASE_AWS_ACCESS_KEY_ID}\naws_secret_access_key=${S3_RELEASE_AWS_SECRET_ACCESS_KEY}\n" >> ~/.aws/credentials
+  pre_start_simulator:
+    description: >-
+      pre start simulator, build may fail is simulator is not started
```
Comment (kvasukib) [inline .circleci/config.yml]: is it possible to wait until the simulator comes up?

File: .circleci/config.yml (hunk 4 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: here only 10.1 and 10.2 work.

File: .circleci/config.yml (hunk 5 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: the command used here will wait  until the simulator comes up

File: .circleci/config.yml (hunk 1 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 3 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 4 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 2 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done

**Pull request merged**